org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPDebugTarget.java

```diff
@@ -632,8 +587,24 @@ public void exited(ExitedEventArguments args) {
 
 	@Override
 	public void thread(ThreadEventArguments args) {
-		refreshThreads.set(true);
-		fireChangeEvent(DebugEvent.CONTENT);
+		triggerUpdateThreads();
+	}
+
+	private CompletableFuture<?> triggerUpdateThreads() {
+		return getDebugProtocolServer().threads().thenAcceptAsync(threadsResponse -> {
+			var threadIds = Arrays.stream(threadsResponse.getThreads()).map(Thread::getId).collect(Collectors.toSet());
+			if (threads.keySet().removeIf(Predicate.not(threadIds::contains))) {
+				fireChangeEvent(DebugEvent.CONTENT);
+			}
+			for (Thread thread : threadsResponse.getThreads()) {
+				DSPThread dspThread = threads.get(thread.getId());
+				if (dspThread == null) {
+					dspThread = new DSPThread(this, thread.getId());
```

Comment:
I may be missing something here, but when is this new dspThread added to threads?